src/Jellyfin.LiveTv/TunerHosts/M3UTunerHost.cs

```diff
@@ -106,11 +116,26 @@ protected override async Task<ILiveStream> GetChannelStream(TunerHostInfo tunerH
                     .SendAsync(message, cancellationToken)
                     .ConfigureAwait(false);
 
-                response.EnsureSuccessStatusCode();
-
-                if (!_disallowedMimeTypes.Contains(response.Content.Headers.ContentType?.ToString(), StringComparison.OrdinalIgnoreCase))
+                if (response.IsSuccessStatusCode)
+                {
+                    if (!_disallowedMimeTypes.Contains(response.Content.Headers.ContentType?.ToString(), StringComparison.OrdinalIgnoreCase))
+                    {
+                        return new SharedHttpStream(mediaSource, tunerHost, streamId, FileSystem, _httpClientFactory, Logger, Config, _appHost, _streamHelper);
+                    }
+                }
+                else if (response.StatusCode == HttpStatusCode.MethodNotAllowed)
+                {
+                    // Fallback to check path extension when the server does not support HEAD method
+                    // Use UriBuilder to remove all query string as GetExtension will include them when used directly
+                    var extension = Path.GetExtension(new UriBuilder(mediaSource.Path).Path);
+                    if (!_disallowedSharedStreamExtensions.Contains(extension, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return new SharedHttpStream(mediaSource, tunerHost, streamId, FileSystem, _httpClientFactory, Logger, Config, _appHost, _streamHelper);
+                    }
+                }
+                else
                 {
-                    return new SharedHttpStream(mediaSource, tunerHost, streamId, FileSystem, _httpClientFactory, Logger, Config, _appHost, _streamHelper);
+                    response.EnsureSuccessStatusCode();
                 }
```

Comment:
```c#
if (response.StatusCode == HttpStatusCode.MethodNotAllowed)
{
	// Fallback to check path extension when the server does not support HEAD method
	// Use UriBuilder to remove all query string as GetExtension will include them when used directly
	var extension = Path.GetExtension(new UriBuilder(mediaSource.Path).Path);
	if (!_disallowedSharedStreamExtensions.Contains(extension, StringComparison.OrdinalIgnoreCase))
	{
		return new SharedHttpStream(mediaSource, tunerHost, streamId, FileSystem, _httpClientFactory, Logger, Config, _appHost, _streamHelper);
	}
}

response.EnsureSuccessStatusCode();

if (!_disallowedMimeTypes.Contains(response.Content.Headers.ContentType?.ToString(), StringComparison.OrdinalIgnoreCase))
{
	return new SharedHttpStream(mediaSource, tunerHost, streamId, FileSystem, _httpClientFactory, Logger, Config, _appHost, _streamHelper);
}
```

Reply:
This won't work, because we want to return the LiveStream if the file format is not allowed, instead of just throw an exception.